teslajsonpy/controller.py

```diff
@@ -406,61 +379,81 @@ async def connect(
-            # Set initial values to setup GridPowerSensor & LoadPowerSensor
-            # Actual values update immediately after setup when refresh is called
-            energysite["grid_power"] = 0
-            energysite["load_power"] = 0
+            for car in self._vehicle_list:
+                vin = car["vin"]
+                if filtered_vins and vin not in filtered_vins:
+                    _LOGGER.debug("Skipping car with VIN: %s", vin)
+                    continue
+
+                self.set_id_vin(car_id=car["id"], vin=vin)
+                self.set_vehicle_id_vin(vehicle_id=car["vehicle_id"], vin=vin)
+                self.__lock[vin] = asyncio.Lock()
+                self.__wakeup_conds[vin] = asyncio.Lock()
+                self._last_update_time[vin] = 0
+                self._last_wake_up_attempt[vin] = 0
+                self._last_wake_up_time[vin] = 0
+                self.__update[vin] = True
+                self.__update_state[vin] = "normal"
+                self.car_state[vin] = car
+                self.set_car_online(vin=vin, online_status=car["state"] == "online")
+                self.set_last_park_time(
+                    vin=vin, timestamp=self._last_attempted_update_time
+                )
+                self.__climate[vin] = {}
+                self.__charging[vin] = {}
+                self.__state[vin] = {}
+                self.__config[vin] = {}
+                self.__driving[vin] = {}
+                self.__gui[vin] = {}
+
+        if self._include_energysites:
+            self._energysite_list = [
+                p
+                for p in self._product_list
+                if p.get(RESOURCE_TYPE) == RESOURCE_TYPE_SOLAR
+                or p.get(RESOURCE_TYPE) == RESOURCE_TYPE_BATTERY
+            ]
 
-            self.__lock[energysite_id] = asyncio.Lock()
-            self._add_energysite_components(energysite)
+            for energysite in self._energysite_list:
+                energysite_id = energysite.get("energy_site_id")
+                battery_id = energysite.get("id")
+
+                if energysite[RESOURCE_TYPE] == RESOURCE_TYPE_SOLAR:
+                    # site_name comes from site_config for non-Powerwall sites
+                    self._site_config[energysite_id] = await self.get_site_config(
+                        energysite_id
+                    )
+                    self._site_data[energysite_id] = {}
+
+                if energysite[RESOURCE_TYPE] == RESOURCE_TYPE_BATTERY:
+                    self._battery_data[energysite_id] = {}
+                    self._battery_summary[energysite_id] = {}
+                # For dealing with sites that always report "Unknown"
+                # Default to True and check during updates
+                self._grid_status_unknown = {energysite_id: True}
+
+                self.__lock[energysite_id] = asyncio.Lock()
+                self.__lock[battery_id] = asyncio.Lock()
```

Comment:
Do we actually need a separate `asyncio.Lock()` based on `battery_id`? I have no experience using `asyncio.Lock()` but just based on the readings, I'm thinking it's not necessary. The only place I use `self.__lock[battery_id]` is with `Controller.update._get_and_process_battery_data` and ``Controller.update._get_and_process_battery_summary` methods, but only because I was basically copying the structure of previous methods.

I think with those two methods, I can just use the `self.__lock` by `energysite_id` that we already have.

Reply:
The lock is for protecting a resource from access if multiple callers may be reading and writing it. Is there a situation where the library may be doing that for energy sites? Like would multiple entities be potentially making calls to teslajsonpy at the same time?

Reply:
I think it's rare but possible. Some folks do have two Tesla solar systems for a single home. This usually happens either if the customer has a very big setup or if they want additional panels after they already had a system installed, they'll just install a separate system so within the Tesla app, you'll see two separate systems.

In this situation, I think it would essentially be like having two cars when it comes to making calls to teslajsonpy. If that's the case, it seems like we could just use `self.__lock[energysite_id]` for all energy site calls but that's what I'm not entirely sure on (if we actually need a separate lock by `battery_id`).

Reply:
I don't know why it was added. For cars it's because all cars are updated every time we hit the end point. I wanted to avoid multiple cars triggering multiple overlapping updates. Maybe locks aren't needed anymore.

Reply:
Ah okay, that makes sense for cars. I'm actually not sure what the behavior is if someone has two systems. I just assumed each system would have its own `energysite_id`, and each time we hit the end point we're specifying the `energysite_id`. I think I'm just going to remove them for now since 1) it's a pretty rare case for solar setups and 2) based on the current (assumed) understanding, we would have to hit each endpoint anyways to get the data for each system. Appreciate the info!